api/glossary_usefulconcepts.md

```diff
@@ -111,46 +145,69 @@ Relational databases set up relationships between items efficiently, and each da
-| |No duplicates, useful for data validation|
-| |Useful to place additional unique conditions on columns|
-|**Composite keys:**| |
-|a.k.a. Compound, composite keys| |
-| More than one columns used to identify a unique value|
+Keys are fundamental in the creation of relationships between data (intra- and inter-tables). Further, they assist in 
+maintaining uniqueness in a table; ensure consistent and valid data as well as allow for efficient data retrieval.
+- **Candidate keys:**
+   - There are multiple keys in a database, one of which can be a primary key
+
+- **Primary key:**
+    - Uniquely identifies each record
+    - Cannot be NULL
+    - The primary key ensures there are no duplicate records within the table (entity integrity)
+    
+- **Alternate keys:**
+    - These are the remaining candidate keys, none of which are currently selected as the primary key
+
+- **Unique keys:**
+    - This key is used to place additional unique conditions on columns
+    - No duplicates in unique keys, and these are useful for data validation
+    - Similar to primary keys, but allow for one NULL value
+
+- **Composite keys:**
+    - a.ka. compound keys
+    - Composite keys are used when more than column is used to identify a value  
 
 ### Data types
 These define the type of value that can be stored in a column. 
 This summary of [PostgreSQL data types](https://www.postgresql.org/docs/9.5/datatype.html) will provide more information.
 - Numeric: 
     - float, real, integers (no decimal, varying precision), decimal (fixed precision (p,s))
-- Characters:
+- Strings:
     - char, var_char, text
 - Boolean: 
     - TRUE / FALSE
-- JSON:
-    - JavaScript Object Notation stores and exchanges data consisting of data types that are arrays or of attribute-value type. 
-- XML:
-    - Extensible markup language (creates information formats and electronically shares structured data via public and corporate networks)
-
-## Universal Resource Locators
-Protocols to locate and access information. These are used for communication between servers and clients. Examples include: [**HTTP**](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol) (hypertext transfer protocols) and [FTP](https://en.wikipedia.org/wiki/File_Transfer_Protocol) (file transfer protocol for file transfers between client and server)
-The general syntax is as you are used to seeing in your browser line:
+- [JSON](https://en.wikipedia.org/wiki/JSON):
+    - JavaScript Object Notation stores and exchanges data consisting of data types that are arrays or of 
+    attribute-value type. 
+- [BLOB](https://en.wikipedia.org/wiki/Binary_large_object):
+    - Binary Large OBject
+    - Collection of [binary data](https://en.wikipedia.org/wiki/Binary_data) stored as a single entity in a database
+     management system (in the case of Baobab, **PostgreSQL**)
+    - These are typically images, audio or multimedia objects  
+
+## Web protocols
+### Universal Resource Locators (URLs)
```

Comment:
I'd remove this "Universal Resource Locators" heading, and move the "Two prominent examples are HTTP ... " here - because those are examples of web protocols, not of URLs.